src/VisualStudio/TestUtilities/Window/EditorWindow.cs

```diff
@@ -1,30 +1,279 @@
+            dteFind.MatchInHiddenText = true;
+            dteFind.MatchWholeWord = true;
+            dteFind.PatternSyntax = vsFindPatternSyntax.vsFindPatternSyntaxLiteral;
+            dteFind.Target = vsFindTarget.vsFindTargetCurrentDocument;
+
+            var findResult = dteFind.Execute();
+
+            if (findResult != vsFindResult.vsFindResultFound)
+            {
+                throw new Exception($"The specified text was not found. ExpectedText: '{expectedText}'; ActualText: '{Text}'");
+            }
+        }
+
+        public void PlaceCursor(string marker) => Find(marker);
+
+        public void TypeText(string text, int wordsPerMinute = 35)
+        {
+            Activate();
+            SendKeys(text.Replace("\r\n", "\r").Replace("\n", "\r"), wordsPerMinute);
+        }
+
+        private void SendKeys(string text, int wordsPerMinute)
+        {
+            var charactersPerSecond = (wordsPerMinute * 4.5) / 60;
+            var delayBetweenCharacters = (int)((1 / charactersPerSecond) * 1000);
+
+            foreach (var character in text)
+            {
+                User32.BlockInput(true);
+
+                var foregroundWindowHandle = User32.GetForegroundWindow();
+
+                var activeWindowHandle = (IntPtr)(_visualStudioInstance.Dte.ActiveWindow.HWnd);
+
+                if (activeWindowHandle == IntPtr.Zero)
+                {
+                    activeWindowHandle = (IntPtr)(_visualStudioInstance.Dte.MainWindow.HWnd);
+                }
+
+                IntegrationHelper.SetFocus(activeWindowHandle);
+
+                var vk = User32.VkKeyScan(character);
+
+                if (vk == -1)
+                {
+                    SendCharacter(character);
+                }
+                else
+                {
+                    if ((vk & 0x0100) != 0)  // SHIFT
+                    {
+                        SendKeyDown(User32.VK_SHIFT);
+                    }
+
+                    if ((vk & 0x0200) != 0)  // CTRL
+                    {
+                        SendKeyDown(User32.VK_CONTROL);
+                    }
+
+                    if ((vk & 0x0400) != 0)  // ALT
+                    {
+                        SendKeyDown(User32.VK_MENU);
+                    }
+
+                    SendKey((ushort)(vk & 0xFF));
+
+                    if ((vk & 0x0100) != 0)  // SHIFT
+                    {
+                        SendKeyUp(User32.VK_SHIFT);
+                    }
+
+                    if ((vk & 0x0200) != 0)  // CTRL
+                    {
+                        SendKeyUp(User32.VK_CONTROL);
+                    }
+
+                    if ((vk & 0x0400) != 0)  // ALT
+                    {
+                        SendKeyUp(User32.VK_MENU);
+                    }
+                }
+
+                IntegrationHelper.SetFocus(foregroundWindowHandle);
+                User32.BlockInput(false);
+
+                Thread.Sleep(delayBetweenCharacters);
```

Comment:
Should this be async? You're trying hard enough to use it in other places, so it seems odd we're not using it here.